yastd/lib/yast2/dbus/installer.rb

```diff
@@ -23,163 +23,67 @@
 
 module Yast2
   module DBus
-    # YaST D-Bus object (/org/opensuse/YaST/Installer)
+    # YaST D-Bus object (/org/opensuse/YaST/Installer1)
     #
     # @see https://rubygems.org/gems/ruby-dbus
     class Installer < ::DBus::Object
       attr_reader :installer, :logger
 
-      PROPERTY_INTERFACE = "org.freedesktop.DBus.Properties"
-      INSTALLER_INTERFACE = "org.opensuse.YaST.Installer"
-
       # @param installer [Yast2::Installer] YaST installer instance
       # @param args [Array<Object>] ::DBus::Object arguments
       def initialize(installer, logger, *args)
         @installer = installer
         @logger = logger
-
-        installer.on_status_change do |status|
-          self.StatusChanged(status.id)
-        end
+        @available_languages = installer.languages.map { |k,v| [k, v.first, {}] }
+        @logger.debug "Available languages #{@available_languages.inspect}"
+        @available_base_products = installer.products
 
         installer.dbus_obj = self
 
         super(*args)
       end
 
-      dbus_interface INSTALLER_INTERFACE do
-        dbus_method :GetLanguages, "out langs:a{sas}" do
-          logger.info "GetLanguages"
-
-          [installer.languages]
-        end
-
-        dbus_method :GetProducts, "out products:aa{ss}" do
-          logger.info "GetProducts"
-
-          products = installer.products.map do |product|
-            { "name" => product.name, "display_name" => product.display_name }
-          end
-          [products]
-        end
-
-        dbus_method :GetStorage, "out proposal:aa{ss}" do
-          logger.info "GetStorage"
-
-          proposal = installer.storage_proposal.filesystems.map do |fs|
-            blk_device = fs.blk_devices.first
-            {
-              "mount"  => fs.mount_point&.path,
-              "device" => blk_device.name,
-              "type"   => fs.type&.to_s,
-              "size"   => blk_device.size.to_i.to_s
-            }
-          end
-          [proposal]
-        end
-
-        dbus_method :GetDisks, "out disks:aa{ss}" do
-          logger.info "GetDisks"
-
-          disks = installer.disks.map do |disk|
-            {
-              "name"  => disk.name,
-              "model" => disk.model,
-              "size"  => disk.size.to_human_string
-            }
-          end
+      LANGUAGE_INTERFACE = "org.opensuse.YaST.Installer1.Language"
```

Comment:
Not directly related to this PR, but I wonder if we could organize the interfaces in a different way. Otherwise, this file is going to grow a lot :-)

Reply:
yeap, @mvidner any hints? Maybe mixins for each interface?